sdk/core/core-client/review/core-client.api.md

```diff
@@ -365,12 +372,30 @@ export interface SimpleMapperType {
     name: "Base64Url" | "Boolean" | "ByteArray" | "Date" | "DateTime" | "DateTimeRfc1123" | "Object" | "Stream" | "String" | "TimeSpan" | "UnixTime" | "Uuid" | "Number" | "any";
 }
 
+export { Span }
+
 // @public
 export interface SpanConfig {
     namespace: string;
     packagePrefix: string;
 }
 
+// @public
+export const SpanKind: {
```

Comment:
I made a conscious decision to both declare these as objects as well as not re-export them from core-tracing. My reasoning for these was:

- I wanted objects instead of enums
- I did not want to make that change in core-tracing to avoid any breaking changes going from enum to object

Reply:
I still think we could have these live in core-tracing, even if it is under a different name and exported with this name from `core-client`.

Reply:
did you consider making it an alternative of strings? e.g. `export const SpanKind = "INTERNAL" | "SERVER" | "CLIENT" | "PRODUCER" | "CONSUMER"`. It is easier on the eye and does not have the enum compat issues, see [here](https://www.typescriptlang.org/play?#code/C4TwDgpgBAymCGA7A0gS0QEwIxQLxQCIBJAOQBUBRAJRIEEAZAqAH0JmoDVqnWCBhekQrkehAApUA8gBEAqn24tCfSSRiyAstwDcAWABQB0JFgIU6DACY8hUpRoNRBdlS5UnAoSKUEJM+Yq8KmqaOgYGAMYA9ogAzsBQAB4AXKZIaJg4+MTk1HSMevoAZgCuiBHAqDFQRQAUIKlw6RaWAJSNZhlWUADeUABOEMAl-YhJ2lAAvpEx8VANaeaY1vh1ia2FBujAEP1F8BHQABo4PQZQF1CxZh3NmQbThvrbu-uHUEfWEIk7mLEfp3Ol2uSFuSysDyAA).

Reply:
> I still think we could have these live in core-tracing, even if it is under a different name and exported with this name from `core-client`.

That makes sense. If we go that route maybe we can just re-export them verbatim today and figure out if we need to do anything special about them once packages no longer use @azure/core-tracing directly



> did you consider making it an alternative of strings? e.g. `export type SpanKind = "INTERNAL" | "SERVER" | "CLIENT" | "PRODUCER" | "CONSUMER"`. It is easier on the eye and does not have the enum compat issues, see [here](https://www.typescriptlang.org/play?#code/C4TwDgpgBAymCGA7A0gS0QEwIxQLxQCIBJAOQBUBRAJRIEEAZAqAH0JmoDVqnWCBhekQrkehAApUA8gBEAqn24tCfSSRiyAstwDcAWABQB0JFgIU6DACY8hUpRoNRBdlS5UnAoSKUEJM+Yq8KmqaOgYGAMYA9ogAzsBQAB4AXKZIaJg4+MTk1HSMevoAZgCuiBHAqDFQRQAUIKlw6RaWAJSNZhlWUADeUABOEMAl-YhJ2lAAvpEx8VANaeaY1vh1ia2FBujAEP1F8BHQABo4PQZQF1CxZh3NmQbThvrbu-uHUEfWEIk7mLEfp3Ol2uSFuSysDyAA).

Yeah, I think that's much nicer for sure! Unfortunately we don't actually own the `Span` object that these are set on, they come directly from OTel tracers. So, they need to be compatible with their SpanKind and SpanStatusCode which are number-based enums

Reply:
> Yeah, I think that's much nicer for sure! Unfortunately we don't actually own the Span object that these are set on, they come directly from OTel tracers. So, they need to be compatible with their SpanKind and SpanStatusCode which are number-based enums

@maorleger but we can compile values between the proposed type and the expected type in OTel in core-tracing, right?

Reply:
I am now re-exporting these from core-tracing